helper/src/network.rs

```diff
@@ -38,3 +44,29 @@ impl Network {
         }
     }
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct ParseNetworkError;
+
+impl FromStr for Network {
+    type Err = ParseNetworkError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "mainnet" => Ok(Self::Mainnet),
+            "testnet" => Ok(Self::Testnet),
+            "stagenet" => Ok(Self::Stagenet),
+            _ => Err(ParseNetworkError),
```

Comment:
really nit but could we add `main` `test` and `stage` as possible values ?

Reply:
We could but then we would also have to allow it in the clap args, which would then display 6 possible values to users which could be confusing.